File: src/gatling/scala/org/openapitools/client/model/QuantumDTO.scala

package org.openapitools.client.model


case class QuantumDTO (
    /* Минимальное количество единиц товара в заказе. Например, если указать 10, покупатель сможет добавить в корзину не меньше 10 единиц.  ⚠️ Если количество товара на складе меньше заданного, ограничение не сработает и покупатель сможет его заказать.  */
    _minQuantity: Option[Integer],
    /* На сколько единиц покупатель сможет увеличить количество товара в корзине.  Например, если задать 5, покупатель сможет добавить к заказу только 5, 10, 15, ... единиц товара.  ⚠️ Если количество товара на складе не дотягивает до кванта, ограничение не сработает и покупатель сможет заказать количество, не кратное кванту.  */
    _stepQuantity: Option[Integer]
)
object QuantumDTO {
    def toStringBody(var_minQuantity: Object, var_stepQuantity: Object) =
        s"""
        | {
        | "minQuantity":$var_minQuantity,"stepQuantity":$var_stepQuantity
        | }
        """.stripMargin
}

File: src/main/java/org/openapitools/server/api/model/QuantumDTO.java
package org.openapitools.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Настройка продажи квантами.  Чтобы сбросить установленные ранее значения, передайте пустой параметр &#x60;quantum&#x60;.  {% cut \&quot;Пример\&quot; %}  &#x60;&#x60;&#x60;json {   \&quot;offers\&quot;: [     {       \&quot;offerId\&quot;: \&quot;08e35dc1-89a2-11e3-8055-0015e9b8c48d\&quot;,       \&quot;quantum\&quot;: {}     }   ] } &#x60;&#x60;&#x60;  {% endcut %} 
 **/
@JsonInclude(JsonInclude.Include.NON_NULL)
public class QuantumDTO   {
  
  private Integer minQuantity;
  private Integer stepQuantity;

  public QuantumDTO () {

  }

  public QuantumDTO (Integer minQuantity, Integer stepQuantity) {
    this.minQuantity = minQuantity;
    this.stepQuantity = stepQuantity;
  }

    
  @JsonProperty("minQuantity")
  public Integer getMinQuantity() {
    return minQuantity;
  }
  public void setMinQuantity(Integer minQuantity) {
    this.minQuantity = minQuantity;
  }

    
  @JsonProperty("stepQuantity")
  public Integer getStepQuantity() {
    return stepQuantity;
  }
  public void setStepQuantity(Integer stepQuantity) {
    this.stepQuantity = stepQuantity;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    QuantumDTO quantumDTO = (QuantumDTO) o;
    return Objects.equals(minQuantity, quantumDTO.minQuantity) &&
        Objects.equals(stepQuantity, quantumDTO.stepQuantity);
  }

  @Override
  public int hashCode() {
    return Objects.hash(minQuantity, stepQuantity);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class QuantumDTO {\n");
    
    sb.append("    minQuantity: ").append(toIndentedString(minQuantity)).append("\n");
    sb.append("    stepQuantity: ").append(toIndentedString(stepQuantity)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
